mantle/go.mod

```diff
@@ -1,56 +1,107 @@
 module github.com/coreos/mantle
 
-go 1.12
+go 1.17
 
 require (
-	github.com/Azure/azure-sdk-for-go v8.1.0-beta+incompatible
-	github.com/Azure/go-autorest v9.1.0+incompatible
+	cloud.google.com/go v0.99.0 // indirect
+	github.com/Azure/azure-sdk-for-go v16.2.1+incompatible
+	github.com/Azure/go-autorest v14.2.0+incompatible
 	github.com/IBM-Cloud/bluemix-go v0.0.0-20210419045805-b50610722085
 	github.com/IBM/ibm-cos-sdk-go v1.6.1
 	github.com/Microsoft/azure-vhd-utils v0.0.0-20161127050200-43293b8d7646
 	github.com/aliyun/alibaba-cloud-sdk-go v1.61.1442
 	github.com/aliyun/aliyun-oss-go-sdk v2.0.3+incompatible
 	github.com/aws/aws-sdk-go v1.34.28
 	github.com/baiyubin/aliyun-sts-go-sdk v0.0.0-20180326062324-cfa1a18b161f // indirect
+	github.com/clarketm/json v1.14.1 // indirect
+	github.com/containerd/stargz-snapshotter/estargz v0.10.1 // indirect
 	github.com/coreos/butane v0.14.0
 	github.com/coreos/coreos-assembler-schema v0.0.0-00010101000000-000000000000
+	github.com/coreos/go-json v0.0.0-20211020211907-c63f628265de // indirect
 	github.com/coreos/go-semver v0.3.0
 	github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
-	github.com/coreos/go-systemd/v22 v22.0.0
+	github.com/coreos/go-systemd/v22 v22.3.2
 	github.com/coreos/ignition/v2 v2.13.0
 	github.com/coreos/ioprogress v0.0.0-20151023204047-4637e494fd9b
 	github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
-	github.com/coreos/stream-metadata-go v0.1.7
+	github.com/coreos/stream-metadata-go v0.0.0-20220315134845-9dfda84d1984
+	github.com/coreos/vcontext v0.0.0-20211021162308-f1dbbca7bef4
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
 	github.com/digitalocean/go-libvirt v0.0.0-20200810224808-b9c702499bf7 // indirect
 	github.com/digitalocean/go-qemu v0.0.0-20200529005954-1b453d036a9c
 	github.com/digitalocean/godo v1.33.0
 	github.com/dimchansky/utfbom v1.1.1 // indirect
+	github.com/docker/cli v20.10.12+incompatible // indirect
+	github.com/docker/distribution v2.7.1+incompatible // indirect
+	github.com/docker/docker v20.10.12+incompatible // indirect
+	github.com/docker/docker-credential-helpers v0.6.4 // indirect
+	github.com/dustin/go-humanize v1.0.0 // indirect
+	github.com/godbus/dbus/v5 v5.0.4 // indirect
+	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/google/go-containerregistry v0.8.0
```

Comment:
Offhand this library looks sane, but it also duplicates a lot of what we already have vendored in the repositories from github.com/containers.

(Now personally, I would say we should stop vendoring all of podman in this repo, but that's a different discussion)

A possible alternative is to fork off `buildah`; we already do that for the other containers.

One specific reason to do that instead of using this library is to ensure that we have a single code path for e.g. authentication, and fewer places we need to change for things like "upload in oci format" (xref https://github.com/coreos/coreos-assembler/pull/2726 ).  I'm guessing go-containerregistry only does v2s2?

Reply:
But to be clear, I'm OK with this as is.

Reply:
> (Now personally, I would say we should stop vendoring all of podman in this repo, but that's a different discussion)
> 
> A possible alternative is to fork off `buildah`; we already do that for the other containers.

If you do that already I can definitely change it. I missed that you already build containers in go in coreos-assembler.

Reply:
> But to be clear, I'm OK with this as is.

Oh nice :)